datafusion/sql/src/expr/mod.rs

```diff
@@ -943,11 +950,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             ) => GetFieldAccess::NamedStructField {
                 name: ScalarValue::from(s),
             },
-            SQLExpr::JsonAccess {
-                left,
-                operator: JsonOperator::Colon,
-                right,
-            } => {
+            SQLExpr::JsonAccess { value, path } => {
```

Comment:
@alamb @jmhain, do you have ideas on how we should modify this branch now?

Previously, we handled `xxx::yyy` in this branch. But now, `JsonOperator::Colon` no longer exists.

It seems to me that we should handle `path.len() == 1` or `path.len() == 2` and build a `GetFieldAccess::ListRange` as the original logic?

Reply:
I pushed f29af9d2ae19c4209d39f90cd7c5d6331a1eb6bd to try it out.

The rest agg parts look much wider that I may hardly handle it by myself in a short time ..